### PR TITLE
Test-Vector 'Disputes'

### PR DESCRIPTION
**Pull request by xDimon**
Related https://github.com/w3f/jamtestvectors/pull/9

### CODE REVIEW

**Comment by turuslan** [comment]
Suggestions [suggest/disputes](https://github.com/qdrvm/cpp-jam/compare/test/disputes...suggest/disputes):
- use `append` instead of `scale::encode` for payload construction (or may use `frown`)
- (TODO: qtils) std hash for bytes
- (TODO: qtils) c++23 ranges contains
- range constructor for `vector` and `set`
- deduplicate multimap iterator
- return `bool` from `jam::ed25519::verify`
- remove unused

**Review by turuslan — APPROVED** [review]

File: test-vectors/disputes/disputes.hpp [hunk 1]
```diff
@@ -0,0 +1,511 @@
...
+    // judgment found to be a contradiction to a work-report’s validity.
+    // Signature related judgment offenders.
+    const auto &faults = ext.faults;
+
+    types::EpochIndex current_epoch = state.tau / config.epoch_length;
+
+    // к - kappa, aka validator set of current epoch
+    const auto &current_epoch_validator_set = state.kappa;
+
+    types::EpochIndex previous_epoch = current_epoch - 1;
```
[the remark this answers is not in the document]
Comment (xDimon) [reply test-vectors/disputes/disputes.hpp:132]: No, it's impossible situation. But we can use copy of epoch as previous one.

File: test-vectors/disputes/disputes.hpp [hunk 2]
```diff
@@ -0,0 +1,511 @@
...
+    // a work-report found to be invalid. Offenders associated with the
+    // guarantee of the report
+    const auto &culprits = ext.culprits;
+
+    // f - set of proofs of one or more validators misbehavior by signing a
+    // judgment found to be a contradiction to a work-report’s validity.
+    // Signature related judgment offenders.
+    const auto &faults = ext.faults;
+
+    types::EpochIndex current_epoch = state.tau / config.epoch_length;
```
[the remark this answers is not in the document]
Comment (xDimon) [reply test-vectors/disputes/disputes.hpp:127]: This type exported from asn-generated types

File: python/asn1.py [hunk 3]
```diff
@@ -190,16 +192,22 @@ def asn_member(t):
...
+            ty.decl = c_using(tname, "uint16_t")
+            continue
         if tname == "U32":
             ty.decl = c_using(tname, "uint32_t")
             continue
         if t["type"] == "NULL":
             t = dict(type="SEQUENCE", members=[])
         if t["type"] == "CHOICE":
-            if tname == "MmrPeak":
+            if tname == "MmrPeak" or tname == "AvailabilityAssignmentItem": # TODO: Need to make in universal
```
Comment (kamilsa) [inline python/asn1.py:210]: Please create issue, resolve, or remove

**Review by kamilsa — APPROVED** [review]

**Pull request merged**